JSQMessagesViewController/Views/JSQMessagesCollectionView.m

```diff
@@ -194,4 +211,57 @@ - (void)messagesCollectionViewCellDidTapAccessoryButton:(JSQMessagesCollectionVi
     [self.accessoryDelegate messageView:self didTapAccessoryButtonAtIndexPath:indexPath];
 }
 
+#pragma mark - Editing overlay view delegate
+
+- (void)editOverlayView:(JSQMessagesEditCollectionOverlayView *)overlayView activated:(BOOL)activated
+{
+    NSIndexPath *indexPath = overlayView.indexPath;
+    if (indexPath == nil) {
+        return;
+    }
+
+    [self.delegate collectionView:self editingOverlayAtIndexPath:indexPath becomeSelected:activated];
+}
+
+- (void)layoutWillDeleteItemAtIndexPath:(NSIndexPath *)indexPath
+{
+    if (indexPath == nil) {
+        return;
+    }
+
+    [self.delegate collectionView:self editingOverlayAtIndexPath:indexPath becomeSelected:NO];
+}
+
+- (void)layoutWillMoveItemAtIndexPath:(NSIndexPath *)indexPath toIndexPath:(NSIndexPath *)newIndexPath
+{
+    if (indexPath == nil || newIndexPath == nil) {
+        return;
+    }
+
+    //this won't cover 100% of cases (like when user rearranging selected items)
```

Comment:
Hmm, how would user rearrange selected items? The only time I ran into a problem with selected index paths was when using load earlier messages, which I resolved with my change to use `isEditableItemSelectedAtIndexPath:`, maybe this covered it?